sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/BlobClientBase.java

```diff
@@ -302,16 +307,32 @@ public BlobInputStream openInputStream(BlobInputStreamOptions options) {
         options = options == null ? new BlobInputStreamOptions() : options;
         ConsistentReadControl consistentReadControl = options.getConsistentReadControl() == null
             ? ConsistentReadControl.ETAG : options.getConsistentReadControl();
-
-        BlobProperties properties = getPropertiesWithResponse(options.getRequestConditions(), null, null).getValue();
-        String eTag = properties.getETag();
-        String versionId = properties.getVersionId();
+        BlobRequestConditions requestConditions = options.getRequestConditions() == null
+            ? new BlobRequestConditions() : options.getRequestConditions();
 
         BlobRange range = options.getRange() == null ? new BlobRange(0) : options.getRange();
         int chunkSize = options.getBlockSize() == null ? 4 * Constants.MB : options.getBlockSize();
 
-        BlobRequestConditions requestConditions = options.getRequestConditions() == null
-            ? new BlobRequestConditions() : options.getRequestConditions();
+        com.azure.storage.common.ParallelTransferOptions pOptions =
+            new com.azure.storage.common.ParallelTransferOptions().setBlockSizeLong((long) chunkSize);
+        BiFunction<BlobRange, BlobRequestConditions, Mono<BlobDownloadAsyncResponse>> downloadFunc =
+            (r, conditions) -> client.downloadWithResponse(r, null, conditions, false);
+        Tuple3<Long, BlobRequestConditions, BlobDownloadAsyncResponse> tuple =
+            ChunkedDownloadUtils.downloadFirstChunk(range, pOptions, requestConditions, downloadFunc, true).block();
+        if (tuple == null) {
+            throw logger.logExceptionAsError(new IllegalStateException("Downloading first chunk returned null"));
+        }
+
+        BlobDownloadAsyncResponse downloadResponse = tuple.getT3();
+        ByteBuffer initialBuffer = FluxUtil.collectBytesInByteBufferStream(downloadResponse.getValue())
+            .map(ByteBuffer::wrap).block();
```

Comment:
Are we collecting Flux of ByteBuffer into single ByteBuffer here?
If so then I'd rather collect the Flux into collection and pass it around instead to avoid mem copies until deemed necessary.

Reply:
I'd rather not take action on. I don't think patterns around InputStreams will cause that to result in memory pressure. 1) It's only 4Mb and 2) I expect people will be reading all or most of that data immediately after opening the stream (when we buffer it). It'd really only be an issue if people were opening but not reading lots of streams, which seems like anti pattern. Or like just not even a pattern